roles/edpm_ovs/tasks/user.yml

```diff
@@ -0,0 +1,11 @@
+---
+- name: Configure openvswitch user and group on the host
+  ansible.builtin.import_role:
+    name: edpm_users
+  vars:
+    edpm_users_users:
+      # 42476 is matching with the uid and gid created by kolla in the openstack containers
+      - {"name": "openvswitch", "uid": "42476", "gid": "42476", "shell": "/sbin/nologin", groups: "hugetlbfs", "comment": "openvswitch user"}
```

Comment:
This has broken the baremetal job. It's trying to change the uid when `openvswitch` has been already installed and running and openvswitch user has a different uid (which is the case with the edpm images used in baremetal job). It fails with the below error. This PR has passed CI, so I guess the job is not using images built with the PR. 

https://logserver.rdoproject.org/14/614/91067d34d1c3d575e9f1151d3b043dcf5b404911/github-check/cifmw-crc-podified-edpm-baremetal/c8d647b/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/pods/configure-network-openstack-edpm-ipam-942ls/logs/openstackansibleee.log

```
TASK [osp.edpm.edpm_users : Ensure user is present on the host [ openvswitch ]] ***
Monday 22 January 2024  10:19:04 +0000 (0:00:03.313)       0:00:06.826 ******** 
[0;31mfatal: [edpm-compute-0]: FAILED! => {"changed": false, "msg": "usermod: user openvswitch is currently used by process 1191\n", "name": "openvswitch", "rc": 8}[0m
```
`openvswitch:x:987:42476:Open vSwitch Daemons:/:/sbin/nologin
`

Reply:
There is a fix https://github.com/openstack-k8s-operators/edpm-ansible/pull/540/files
and it seems I should switch it back to using try/except blocks?

Reply:
> which is the case with the edpm images used in baremetal job

Shouldn't also that BM job using that same ID, that's likely originates from kolla users mappings?

Reply:
> There is a fix https://github.com/openstack-k8s-operators/edpm-ansible/pull/540/files

I don't think the change here is relevant to the issue. AFAIK user would be created during installation and edpm image used in baremetal job already installs openvswitch[1]. IMO we can get rid of the user creation completely here.

[1] https://github.com/openstack-k8s-operators/edpm-image-builder/blob/main/images/edpm-hardened-uefi-centos-9-stream.yaml#L21

Reply:
we shoudl not be installing OVS in the baremetal image as not all EDPM nodes should have ovs install. 

so we should remove the ovs installation at https://github.com/openstack-k8s-operators/edpm-image-builder/blob/main/images/edpm-hardened-uefi-centos-9-stream.yaml#L21

Reply:
it looks like the fix i had added to only set the user if its not already created got lost when i removed the last two commits form this pr...

lets proceed with #540

Reply:
> we shoudl not be installing OVS in the baremetal image as not all EDPM nodes should have ovs install.

If it's not needed in the base images, it can be dropped from edpm-image-builder. However, I don't understand why we want to unnecessarily have the check to create the user, as what I can see it would surely be created with when installing the rpm.

[1] https://pkgs.devel.redhat.com/git/rpms/openvswitch/tree/openvswitch.spec?h=rhel-9.0.0#n329

Reply:
if the rpm was correctly pinnign the user to the one expect on OpenStack system you would not have had the ci issue.

the container expect ovs to have a specific user and group that this not the same as the one used by rhel when packaging ovs.

the ovs we install in our openstack distribution comes form the fast datapath repository
and is not part of our openstack repos so it is not packaged to run as the users we use historically in our product when building the container.

as a result we need to ensure we create the user with the expect user id and groups before the rpm is installed.

Reply:
mismatching IDs will hit us if we'd want to use UNIX sockets for ovn/ovs/nova compute interactions, IIUC. But this is good the fix doesn't change the status quo at least

Reply:
the fix helps with "upgrade" of workloads during adoption, where we don't touch OVS bits yet

Reply:
creating the user/group with the uid/group id specified in tcib is not for upgrades
it to fix greenfield to ensure they are assigned.

skipping the user/group creation in #540 is a workaround for the fact the tcib/tripleo uid/gid are out of sync.

we can and likely should correct the uid/gid in tcib 
https://github.com/openstack-k8s-operators/tcib/blob/main/container-images/kolla/base/uid_gid_manage.sh#L62
to match those used in tripleo

https://github.com/openstack/tripleo-common/blob/stable/wallaby/container-images/kolla/base/uid_gid_manage.sh#L66

however looking at stable wallaby we used the same user/group id dere as tcib is using today.

_SUPPORTED_USERS['openvswitch']='openvswitch 42476 42476'

it is also the same in stable/train

https://github.com/openstack/tripleo-common/blob/stable/train/container-images/kolla/base/uid_gid_manage.sh#L65

so either the users in the container and the hsot have always been broken or there is something else happening to correct this that I'm not aware of.